linked_list/implementation.py

```diff
@@ -1,40 +1,180 @@
-from .interface import AbstractLinkedList
-
+from linked_list.interface import AbstractLinkedList
+from linked_list.node import Node
 
 class LinkedList(AbstractLinkedList):
-    """
-    Implementation of an AbstractLinkedList inteface.
-    """
-
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        
+        if elements:
+            for element in elements:
+                self.append(element)
 
+            
     def __str__(self):
-        pass
+        current_node = self.start
+        return_list = []
+        while current_node is not None:
+            return_list.append(current_node.elem)
+            current_node = current_node.next
+        
+        return str(return_list)    
 
-    def __len__(self):
-        pass
 
+    def __len__(self):
+        count = 0
+        if self.start is None:
+            return count
+        else: 
+            current_node = self.start
+            next_node = current_node.next
+            count+=1
+            while next_node is not None:
+                current_node = next_node
+                next_node = next_node.next
+                count += 1
+            return count   
+            
+            
     def __iter__(self):
-        pass
+        current_node = self.start
+        while True:
+            yield current_node
+            current_node = current_node.next
+            if current_node is None :
+                raise StopIteration
+
 
     def __getitem__(self, index):
-        pass
+        if index >= len(self) :
+            raise IndexError
+        
+        if index is None :
+            raise KeyError
+        
+        curr_index = 0
+        for node in self :
+            if curr_index == index : 
+                return node.elem
+            curr_index += 1
+    
 
     def __add__(self, other):
-        pass
-
+        new_Linkedlist = LinkedList()
+        current_node = self.start 
+        other_node = other.start
+        if other is None and self is None :
+            return new_Linkedlist
+        elif other_node is None :
+            return self
+        elif self.start is None :
+            return other
+        else:
+            count = 0
+            while count < len(self):
+                new_Linkedlist.append(current_node.elem)
+                current_node = current_node.next
+                count += 1
+                
```

Comment:
Instead of using while loops and a counter to iterate through the linked lists you can use your __iter__ method by doing 
`for node in self:`